core/primitives-core/src/serialize.rs

```diff
@@ -212,3 +216,27 @@ pub mod option_u128_dec_format {
         }
     }
 }
+
+#[derive(Debug, Deserialize)]
+pub struct DeserializedWithDebug<T>(pub T);
+
+impl<T> DeserializedWithDebug<T> {
+    fn _deserialize<'de, D>(deserializer: D) -> Result<T, D::Error>
+    where
+        T: Deserialize<'de>,
+        D: Deserializer<'de>,
+    {
+        if cfg!(debug_assertions) {
+            let mut unused_fields = Vec::new();
+            let de_result: Result<T, D::Error> = serde_ignored::deserialize(deserializer, |path| {
+                unused_fields.push(path.to_string());
+            });
+
+            debug!("missed fields while deserializing: {:?}", unused_fields);
```

Comment:
This might want a `target` specific to this functionality and also perhaps a higher level (`info`? `warn`?). That way anybody who doesn't want to see these messages can disable them by setting a larger level filter for that target, while for people who are running `neard` or other utilities for the first time, the message would readily come up.

```suggestion
            info!(target: "near::serialization::missing_fields", message = "unknown fields while deserializing", ?unused_fields);
```

perhaps?